java/ql/lib/semmle/code/java/security/InsecureLdapAuthQuery.qll

```diff
@@ -0,0 +1,72 @@
+/** Provides dataflow configurations to reason about insecure LDAP authentication. */
+
+import java
+import DataFlow
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.security.InsecureLdapAuth
+
+/**
+ * A taint-tracking configuration for `ldap://` URL in LDAP authentication.
+ */
+private module InsecureLdapUrlConfig implements DataFlow::ConfigSig {
+  predicate isSource(DataFlow::Node src) { src.asExpr() instanceof InsecureLdapUrl }
+
+  predicate isSink(DataFlow::Node sink) {
+    exists(ConstructorCall cc |
+      cc.getConstructedType().getAnAncestor() instanceof TypeDirContext and
+      sink.asExpr() = cc.getArgument(0)
```

Comment:
Since this sink definition is reused in the three configurations, how about we model it as a class and just say `sink instanceof InsecureLdapUrlSink` (or a more general name, like `DirContextEnvironment`) in the three of them?